src/components/Chart.tsx

```diff
@@ -74,12 +76,18 @@ export function Chart(props: ChartProps) {
   const createGroupHandlers = (d: CircleOrRectHiearchyNode) =>
     isBlob(d.data)
       ? {
-        onClick: () => navigate(`./details/${global.btoa(JSON.stringify(d.data))}`),
+        onClick: () => {
+          navigate(`./details/${global.btoa(JSON.stringify(d.data))}`);
+          setClickedBlob(d.data as HydratedGitBlobObject);
+        },
         onMouseOver: () => setHoveredBlob(d.data as HydratedGitBlobObject),
         onMouseOut: () => setHoveredBlob(null),
       }
       : {
-        onClick: () => navigate("/repo"),
+        onClick: () => {
+          navigate("/repo");
+          setClickedBlob(null);
```

Comment:
We need to either remove the details route or remove the ClickedBlob state, since these might get out of sync with eachother.